### ISSUE TITLE
[ FF7 ] World model vertex colors appear to be inverted

### ISSUE THREAD

**Issue by HeyItsLollie** [julianxhokaxhiu/FFNx#610]
Vertex colors for World models appear to be inverted, compared to PS1. This issue was stumbled upon while taking screenshots for #609.

These screenshots were taken at Kalm, immediately outside Midgar. Open these images in separate tabs and flip between them, the issue will be immediately apparent.

PlayStation:
![Field_PS1](https://github.com/julianxhokaxhiu/FFNx/assets/30498581/435bad22-32fc-44f4-9f15-835227ad76a5)

Steam. Note that in this example image, I have edited pixel.hlsl at Line 28 to multiply texture color brightness by 2, to make the vertex coloring issue easier to spot in comparisons.
![Field_Steam-Hack](https://github.com/julianxhokaxhiu/FFNx/assets/30498581/99aeef16-a086-43bc-bdc6-95b2170ee732)

And for posterity, here is Steam as it appears normally, without any modifications:
![Field_Steam-Stock](https://github.com/julianxhokaxhiu/FFNx/assets/30498581/f30d6ab1-66a7-4f46-83b7-b4ff12515cb3)

**Comment by julianxhokaxhiu** [comment]
Hi, thanks for the report. Did you manage to reproduce the issue on the latest Canary? If yes, can you please send your `FFNx.toml` file?

**Comment by HeyItsLollie** [comment]
Yep, this is present on FFNx-v1.16.0.61. Here's the FFNx.toml file! (zipped so Github will accept it): [FFNx.zip](https://github.com/julianxhokaxhiu/FFNx/files/12744069/FFNx.zip)

**Comment by CosmosXIII** [comment]
Thanks for the report. Clearly light direction is inverted. Problem is the terrain in world map mode works differently than other 3d models. The terrain geometry is pre-transformed on the CPU making it difficult to fix in a similar way as I did for other models. Of course, we could fix the CPU implementation of the lighting calculations that the game is doing to invert light directions but my focus right now is moving as much lighting calculations to the GPU (shader) rather than keeping it on the CPU. In this sense, can't share the details right now but we have something related to the worldmap coming in a few months that may circumvent this problem.
